R/enrichment_map.R

```diff
@@ -185,6 +221,57 @@ enrichment_map <- function(res_enrich,
   # re-sorting the vertices alphabetically
   rank_gs <- rank(V(emg)$name)
   emg <- permute.vertices(emg, rank_gs)
+  
+  # add communities based on different community detection algorithms
+  add_communities <- function(em) {
+    cd_algorithms <- c(cluster_louvain, 
+                       cluster_edge_betweenness, 
+                       cluster_fast_greedy)
+    names(cd_algorithms) <- c("louvain", 
+                              "in_betweenness", 
+                              "fast_greedy")
```

Comment:
We could have the option that the function is passed directly - this would avoid to restrict the selection to these algos